### PR TITLE
Rename `rust_2015` → `is_rust_2015`

### PR DESCRIPTION
**Pull request by WaffleLapkin**
r? @compiler-errors
https://github.com/rust-lang/rust/pull/107508#discussion_r1092300088

### CODE REVIEW

File: compiler/rustc_parse/src/parser/ty.rs [hunk 1]
```diff
@@ -702,7 +702,7 @@ impl<'a> Parser<'a> {
     /// Is a `dyn B0 + ... + Bn` type allowed here?
     fn is_explicit_dyn_type(&mut self) -> bool {
         self.check_keyword(kw::Dyn)
-            && (!self.token.uninterpolated_span().rust_2015()
+            && (self.token.uninterpolated_span().rust_2018()
```
Comment (eggyal) [inline compiler/rustc_parse/src/parser/ty.rs:705]: Wouldn't it make sense for this similarly to be renamed `is_rust_2018`, and so on for all such edition tests?

Comment (WaffleLapkin) [reply compiler/rustc_parse/src/parser/ty.rs:705]: No, see the linked comment. `rust_2015` is `== 2015`, but `rust_2018` is `>= 2018` (similarly for others).

**Review by davidtwco — APPROVED** [review]

**Comment by compiler-errors** [comment]
Hmm... I think bors is busted. I'll post about it on T-infra.

@bors treeclosed=50

**Pull request merged**